### PR TITLE
Allow the deletion of others' messages

### LINKED ISSUE #1683: [SUGGESTION] Delete others messages
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

WhatsApp has added the feature to delete others messages as an admin, and using message.delete(true) does not seem to work for others messages

### Describe the solution you'd like.

Making message.delete work for other members as long as you have permission to

### Describe an alternate solution.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by soudblox**
Removing the fromMe check in the `message#delete` method does the job

I have tested it and it does indeed work

Does not show in WhatsApp desktop UWP though, because it's not supported there yet

fixes #1683

### CODE REVIEW

**Comment by sergiooak** [comment]
Have you tested? To me it does't work.

File: src/structures/Message.js [hunk 1]
```diff
@@ -436,7 +436,7 @@ class Message extends Base {
         await this.client.pupPage.evaluate((msgId, everyone) => {
             let msg = window.Store.Msg.get(msgId);
 
-            if (everyone && msg.id.fromMe && msg._canRevoke()) {
+            if (everyone && msg._canRevoke()) {
                 return window.Store.Cmd.sendRevokeMsgs(msg.chat, [msg], {type: 'Sender'});
```
Comment (sergiooak) [inline src/structures/Message.js:440]: Change the type between "Sender" and "Admin" depending is `msg.id.fromMe`
`return window.Store.Cmd.sendRevokeMsgs(msg.chat, [msg], {type: msg.id.fromMe ? 'Sender' : 'Admin'});`

**Review by sergiooak — CHANGES_REQUESTED** [review]

**Comment by sergiooak** [comment]
Tested with this recommended change, and now it works
image

File: src/structures/Message.js (hunk 1 above)
Comment (ZeroDark1712) [reply src/structures/Message.js:440]: Captura de pantalla 2022-09-11 145755
Works!

**Review by ZeroDark1712 — APPROVED** [review]

**Comment by soudblox** [comment]
> Have you tested? To me it does't work.

Weird, for me it worked

File: src/structures/Message.js (hunk 1 above)
Comment (soudblox) [reply src/structures/Message.js:440]: > Change the type between "Sender" and "Admin" depending is `msg.id.fromMe` `return window.Store.Cmd.sendRevokeMsgs(msg.chat, [msg], {type: msg.id.fromMe ? 'Sender' : 'Admin'});`

Alright

**Comment by soudblox** [comment]
> Tested with this recommended change, and now it works, still need to implement a validation is chat adm image

I should also add a check if it's a group chat or not

**Review by purpshell — APPROVED** [review]

**Review by tuyuribr — APPROVED** [review]

**Comment by PurpShell** [comment]
> > Tested with this recommended change, and now it works, still need to implement a validation is chat adm image
> 
> I should also add a check if it's a group chat or not
> That would require me to fetch the group first, then the participants to check if the user is a super admin or not

You can simply get the chat (group) and loop through the participants in the browser plane.

**Comment by sergiooak** [comment]
The _canRemoveke do t exclude the need to check if it is a group?

On private the _canRevoke will be false for non sender message anyways.

I suppose that all use cases are already covered with the current if statement

**Comment by soudblox** [comment]
> The _canRemoveke do t exclude the need to check if it is a group?
> 
> On private the _canRevoke will be false for non sender message anyways.
> 
> I suppose that all use cases are already covered with the current if statement

I'm not particularly familiar with the WhatsaApp Web API so I don't really know a lot of private functions that exist

**Comment by soudblox** [comment]
@PurpShell Is there anything else that I have to do?

**Comment by shirser121** [comment]
@PurpShell I think we can merge it

**Pull request merged**